src/pages/reactivistas/index.tsx

```diff
@@ -1,26 +1,127 @@
 import Layout from '../../components/Layout';
-import { createReactGroup, getPersonByDiscordId } from '../../lib/api';
+import { getAllReactGroups } from '../../lib/api';
 import { useForm } from 'react-hook-form';
 import { ReactGroup } from '../../lib/types';
-import { useState } from 'react';
+import { useState, Fragment, FormEvent } from 'react';
 import Modal from '../../components/Modal';
+import { GetStaticProps, InferGetStaticPropsType } from 'next';
+import { usePreviewSubscription } from '../../lib/sanity';
+import { reactGroupQuery } from '../../lib/querys';
+import Hero from '../../components/Hero';
 
-const ReactGroupPage: React.FC = () => {
+const ReactGroupPage: React.FC<
+  InferGetStaticPropsType<typeof getStaticProps>
+> = ({ data, preview }) => {
   const [isModalOpen, setIsModalOpen] = useState(false);
 
+  const { data: groups } = usePreviewSubscription(reactGroupQuery, {
+    initialData: data,
+    enabled: preview,
+  });
+
+  const onAddParticipantSubmit = async (
+    event: FormEvent<HTMLFormElement>,
+    discordUser: string,
+    id: string,
+  ) => {
+    event.preventDefault();
+
+    const data = {
+      discordUser: discordUser,
+      id: id,
+    };
+
+    try {
+      const res = await fetch('/api/add-participant', {
+        method: 'POST',
+        body: JSON.stringify(data),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
   return (
     <Layout title="Iniciativas">
+      <Hero small title="Reactivistas" />
       <div className="pb-24 bg-indigo-100 sm:pt-10">
+        <div className="container mx-auto m-10 p-10 bg-white shadow">
+          <h1 className="text-2xl font-bold leading-7 text-black sm:text-2xl sm:leading-9 sm:truncate">
+            Sumate a los grupos que están comenzando
+          </h1>
+          <div className="flex flex-wrap m-10">
+            {groups.map((group: ReactGroup) => {
+              const [discordUser, setDiscordUser] = useState('');
+
+              return (
+                <Fragment key={group.name}>
+                  {group.status === 'approved' ? (
```

Comment:
`{group.status === 'approved' && (...` would have the same effect and you would skip having to do the `: null` on line 110.

Reply:
Also, can we just do the filter on the Sanity query so we don't even get non-approved groups on client?